src/tests/time-base-api-test/time-base-api-test.c

```diff
@@ -0,0 +1,278 @@
+/*
+ *      Copyright (c) 2020, United States government as represented by the
+ *      administrator of the National Aeronautics Space Administration.
+ *      All rights reserved. This software was created at NASA Goddard
+ *      Space Flight Center pursuant to government contracts.
+ *
+ *      This is governed by the NASA Open Source Agreement and may be used,
+ *      distributed and modified only according to the terms of that agreement.
+ */
+
+/*
+ * Filename: time-base-api-test.c
+ *
+ * Purpose: This file contains functional tests for "osapi-timebase"
+ *
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "common_types.h"
+#include "osapi.h"
+#include "utassert.h"
+#include "uttest.h"
+#include "utbsp.h"
+
+static uint32 TimerSyncCount = 0;
+static uint32 TimerSyncRetVal = 0;
+
+static uint32 UT_TimerSync(uint32 timer_id)
+{
+    ++TimerSyncCount;
+    return TimerSyncRetVal;
+}
+
```

Comment:
Note this sync function is going to immediately return 0, which causes a "spin loop"  in the timebase.  There is some protection against this but its really not something you should do, its just to protect against locking the CPU.